api/settings/v1/settings.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package settings.v1;
+
+service SettingsService {
+    rpc All(AllSettingsRequest) returns (AllSettingsResponse) {}
+    rpc Set(SetSettingsRequest) returns (SetSettingsResponse) {}
+}
+
+message AllSettingsRequest {}
+
+message AllSettingsResponse {
+    repeated Setting settings = 1;
+}
+
+message SetSettingsRequest {
+    Setting setting = 1;
+}
+
+message SetSettingsResponse {
+    Setting setting = 1;
+}
+
+message Setting {
+    string name = 1;
```

Comment:
You'll need optimistic concurrency control here https://en.wikipedia.org/wiki/Optimistic_concurrency_control

To do so you'll need to have a modifiedAt for each setting, that you'll use when modifying, if not equal you'll reject the change it means somebody did a change in between and you're about to erase it.

Reply:
Check out this [here](https://github.com/grafana/pyroscope/pull/2711/files#diff-9260ea2e34adbb6eedcaaf2bf8ff746c592f00ccb0fad752e470f85a133890ceR55-R58). Does that seem about right?

Reply:
Yes that's right, I suggest to make the error more obvious.